packages/config-ui/src/layout/layout-contents.jsx

```diff
@@ -14,21 +14,23 @@ class RawLayoutContents extends React.Component {
 
   render() {
     const { mode, secondary, children, classes } = this.props;
+    const hasSettingsPanel = Object.entries(secondary.props.configuration).some(([propName, obj]) => !!obj?.settings);
```

Comment:
I would add a sanity check here - none of the above should be undefined, but just to make sure...
secondary?.props?.configuration || {}